// @botpress/webchat/dist/index.js
import { A as t, B as r, C as m, f as c, e as i, h as n, F as h, H as p, i as S, L as d, M as l, j as C, k as u, m as P, R as b, W as g, n as v, v as M, w as f, D as k, E as y, p as W, s as x, o as B, b as H, G as I, q as L, l as R, y as T, t as w, x as A, d as F, u as j, z as q } from "./index-45673b38.js";
import "react";
import "react-dom";
export {
  t as Avatar,
  r as Block,
  m as Composer,
  c as ComposerButton,
  i as ComposerInput,
  n as Container,
  h as Fab,
  p as Header,
  S as HeaderRoot,
  d as LoadingIndicator,
  l as Message,
  C as MessageList,
  u as Modal,
  P as ModalProvider,
  b as RestartConversation,
  g as Webchat,
  v as WebchatProvider,
  M as avatarThemeSchema,
  f as blockThemeSchema,
  k as clientModePropsSchema,
  y as configPropsSchema,
  W as configurationSchema,
  x as cssStyleSchema,
  B as descriptionItemSchema,
  H as getClient,
  I as initPropsSchema,
  L as markdownTypeSchema,
  R as renderers,
  T as statePropsSchema,
  w as styleOptionSchema,
  A as themeSchema,
  F as useClient,
  j as useWebchatContext,
  q as userPropsSchema
};
